app_backend/app/routers/favorites.py

```diff
@@ -34,11 +44,17 @@ async def toggle_favorite(resp: Response, favorite: FavoriteFood):
         resp.status_code = status.HTTP_400_BAD_REQUEST
         return {"err": "Empty food_id"}
 
-    existing_food_state = True  # TODO: get food_id's state here
-    if existing_food_state == favorite.state:
+    favorites = user.favorites
+    if (favorite.food_name in favorites) == favorite.state:
         resp.status_code = status.HTTP_200_OK
         return {"msg": "No row is affected."}
-    # TODO: Update food_id here
-    # toggle food_id
+    else:
+        if favorite.state:
+            favorites.append(favorite.food_name)
+
+    # Save
+    user.favorites = favorites
+    user.save()
```

Comment:
add `dry_run` parameter in this method that can be set in the pay_load in tests. We don't want to write data when testing